notification/store.go

```diff
@@ -328,26 +350,12 @@ func (db *DB) FindManyMessageStatuses(ctx context.Context, ids ...string) ([]Mes
 			return nil, err
 		}
 		s.ProviderMessageID = providerMsgID.String
-
-		switch lastStatus {
-		case "queued_remotely", "sending":
-			s.State = MessageStateSending
-		case "pending":
-			s.State = MessageStatePending
-		case "sent":
-			s.State = MessageStateSent
-		case "delivered":
-			s.State = MessageStateDelivered
-		case "failed", "bundled": // bundled message was not sent (replaced) and should never be re-sent
-			// temporary if retry
-			if hasNextRetry {
-				s.State = MessageStateFailedTemp
-			} else {
-				s.State = MessageStateFailedPerm
-			}
-		default:
+		state := GetMessageStatusState(lastStatus, hasNextRetry)
```

Comment:
Just use `s.State` instead of declaring a new variable
```suggestion
		s.State = GetMessageStatusState(lastStatus, hasNextRetry)
```